### ISSUE TITLE
Develop a "run convo" function within the NPC dialogue module

### ISSUE THREAD

**Issue by jeyeong** [uchicago-cs/chiventure#868]
The current dialogue module only has functions for building the data structures used to store conversations. To integrate conversations into chiventure, we will implement functions for traversing dialogue trees, specifically a "run_convo" function which takes in the user's input as a string, moves to the next node, and returns the corresponding dialogue prompt to be printed to the cli.
Assignees: Peter, James
team: npc2

**Comment by jeyeong** [comment]
Wrote a function within the _dialogue_ module that takes a convo_t object and runs the conversation. The function has not been designed to handle inputs from the CLI. This would require deeper integration between the _dialogue_ module and the _CLI_ module, and would likely rely on the implementation of a "conversation mode" in the CLI (see [Issue 864](https://github.com/uchicago-cs/chiventure/issues/864)).

**Comment by jeyeong** [comment]
Wrote and committed this function under the npc/dialogue branch, along with other dialogue functions (see [Issue 888](https://github.com/uchicago-cs/chiventure/issues/888)).

In its current form, the function takes a convo struct and runs the conversation. It accepts user input via scanf() and prints output using printf().

Eventually, the goal will be to accept input via some channel of the CLI, and print output using print_cli().

**Comment by jeyeong** [comment]
After discussing the design of the conversation procedure with @pkasner, wrote a run_conversation_step() function that receives an integer input, together with a convo struct and a char * out-parameter,

**Comment by jeyeong** [comment]
I discussed the implementation of the run_conversation() function with @pkasner, where we deduced it was necessary to build a "stepwise" function that accepts an input (parsed from the CLI) and traverses a single edge to a new node, where it returns the string to be printed out to the CLI.

To that end, I implemented a run_conversation_step() function, whose interface looks like this:
int run_conversation_step(convo_t *c, int input, char *outstring);

For this function to work properly, a slight change in the design of the convo struct is required. Specifically, the addition of a node_t *cur_node member. This will be done in a future sprint. To view the implementation of this function, have a look at the dialogue.c file under the npc/dialogue branch.

**Comment by jeyeong** [comment]
[Pull request](https://github.com/uchicago-cs/chiventure/pull/906) has been made. This issue will be kept open, however, as more work on this function still needs to be done.

**Comment by jeyeong** [comment]
I removed the run_conversation function. The run_conversation_step function has been fully developed to facilitate the integration of the dialogue module with the CLI. I also added a start_conversation function.

**Comment by jeyeong** [comment]
I have made the following changes to the start_conversation() and run_conversation_step() functions:

1. In order to facilitate conditional dialogue, I had to remove the option_number field from the edge list. This affected the way we previously produced the return string, since that implementation made use of option_number field. To solve this, I updated the string creation function to work without the option_number field.
2. In order to facilitate action integration, I added components within the two functions to check for and run actions at the nodes they land on.
